### ISSUE TITLE
--frozen produces a bundle that "can't find executable rake for gem rake"

### ISSUE THREAD

**Issue by lwoggardner** [rubygems/rubygems#3229]
I'm trying to ensure that in my build scripts that the Gemfile.lock does not need updating.  If I understand correctly, --frozen is the correct option to use for that.

```bash
bundle install --binstubs=bin --frozen --without development || exit 2
bundle list
bin/rake --version
```

Produces...
```
ruby 2.3.7p456 (2018-03-28 revision 63024) [x86_64-linux-gnu]
Bundler version 1.16.5
Checking frozen bundle
Fetching gem metadata from https://rubygems.org/.............
Fetching rake 12.3.1
Installing rake 12.3.1
... <more gems>
Bundle complete! 50 Gemfile dependencies, 81 gems now installed.
Gems in the group development were not installed.
Bundled gems are installed into `./vendor/bundle`
...
Gems included by the bundle:
...
* rake (12.3.1)
...

<DIR>/.gem/ruby/2.3/gems/bundler-1.16.5/lib/bundler/rubygems_integration.rb:462:in `block in replace_bin_path': can't find executable rake for gem rake. rake is not currently included in the bundle, perhaps you meant to add it to your Gemfile? (Gem::Exception)
```

If I do not include the --frozen in the bundle install command, then it successfully finds and runs rake.

Am I misusing --frozen somehow?  I have also tried specifically using --path, to no effect.

**Comment by deivid-rodriguez** [comment]
I think this should've been fixed in bundler 2.2.0.rc.1. If you're able to verify, we appreciate it.

**Comment by deivid-rodriguez** [comment]
I'll assume it's fixed, feel free to reopen if you have a repro on the most recent prerelease!
